src/world/Actor/PlayerEvent.cpp

```diff
@@ -145,6 +145,8 @@ void Sapphire::Entity::Player::playGilShop( uint32_t eventId, uint32_t flags,
   openGilShopPacket->data().eventId = eventId;
   openGilShopPacket->data().sceneFlags = flags;
   openGilShopPacket->data().actorId = getId();
+  openGilShopPacket->data().params[ 0 ] = 1;
```

Comment:
did I get the packet wrong? there's a field before this for the arg count

Reply:
this one is tricky. we only need the next param but this one cannot be 0. on retail it's 0x721A64A2 not looking like a count at all, and the count field we have is always 0

Reply:
ok the packet is wrong, see the latest update. I wonder why my wrong packet was working lol

Reply:
yeah okay, maybe there's something else going on there, will need to double check the packet but this is fine for now